packages/medusa/src/repositories/staged-job.ts

```diff
@@ -1,5 +1,29 @@
 import { EntityRepository, Repository } from "typeorm"
-import { StagedJob } from "../models/staged-job"
+import { QueryDeepPartialEntity } from "typeorm/query-builder/QueryPartialEntity"
+import { StagedJob } from "../models"
+import { rowSqlResultsToEntityTransformer } from "../utils"
 
 @EntityRepository(StagedJob)
-export class StagedJobRepository extends Repository<StagedJob> {}
+export class StagedJobRepository extends Repository<StagedJob> {
+  async insertBulk(jobToCreates: QueryDeepPartialEntity<StagedJob>[]) {
+    const queryBuilder = this.createQueryBuilder()
+      .insert()
+      .into(StagedJob)
+      .values(jobToCreates)
+
+    // TODO: remove if statement once this issue is resolved https://github.com/typeorm/typeorm/issues/9850
+    if (queryBuilder.connection.driver.isReturningSqlSupported("insert")) {
+      queryBuilder.returning("*")
+      const rawStagedJobs = await queryBuilder.execute()
+      return rawStagedJobs.generatedMaps
```

Comment:
question: Shouldn't we still use the transformer here? This is the case where `returning` is supported

Reply:
my bad, I forgot the negation